ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/MigrationsDirectoryUtil.java

```diff
@@ -46,6 +48,32 @@
   private MigrationsDirectoryUtil() {
   }
 
+  /**
+   * Returns the migrations directory path based on the provided config and
+   * config file path. This method does not perform validation on the provided
+   * directory location; that responsibility is left to downstream methods.
+   */
+  public static String getMigrationsDir(
+      final String configFilePath,
+      final MigrationConfig config
+  ) {
+    final String migrationsDir = config.getString(MigrationConfig.KSQL_MIGRATIONS_DIRECTORY_PATH);
+    if (migrationsDir != null && !migrationsDir.isEmpty()) {
+      return migrationsDir;
+    } else {
+      return getMigrationsDirFromConfigFile(configFilePath);
+    }
+  }
+
+  /**
+   * Returns the migrations directory path assuming the default file structure
+   * (as created by the 'ksql-migrations new-project' command). This method should
+   * NOT be called directly outside of testing; prefer
+   * {@link MigrationsDirectoryUtil#getMigrationsDir(String, MigrationConfig)}
+   * instead as users can override the default location returned by this method
+   * in their migrations config if they wish.
+   */
+  @VisibleForTesting
   public static String getMigrationsDirFromConfigFile(final String configFilePath) {
```

Comment:
Does this need to be public?

Reply:
Unfortunately, yeah. The only place it's used (besides as a helper method in this file) is from MigrationsTest.java, which is in a different package.